app/assets/javascripts/components/course_creator/course_subject_selector.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import CreatableInput from '../common/creatable_input.jsx';
+
+const CourseSubjectSelector = ({ updateCourse }) => {
+  const [options, setOptions] = useState();
+
+  useEffect(() => {
+    const opts = [];
+
+    fetch('/wizards/researchwrite.json')
+      .then(resp => resp.json())
+      .then((data) => {
+        const jsonObj = data;
+        const jsonObjects = jsonObj[5].options;
```

Comment:
This needs to be documented, as it's not going to be obvious what is going on here and how it is coupled to the researchwrite.yml file; anything that addes to content above the subject choices will require changing the index here.

Reply:
`jsonObjects` is not a very informative variable name here. Maybe `courseSubjects` or `subjectOptions`?

Reply:
Replaced the variable `jsonObjects` into `subjectOptions`.

Reply:
Great. Just needs some inline comments explaining what's going on, especially with `jsonObj[5].options`, and then it will be ready to merge.

Reply:
Thank you I will provide inline comments.